demo/java/src/main/java/io/github/cosinekitty/astronomy/demo/Main.java

```diff
@@ -25,27 +26,14 @@ private static Time parseTime(String args[], int index) {
             Date now = new Date();
             return Time.fromMillisecondsSince1970(now.getTime());
         }
-        Pattern pattern = Pattern.compile("^(\\d{4})-(\\d{2})-(\\d{2})(T(\\d{2}):(\\d{2})(:(\\d{2}(\\.\\d+)?))?Z)$");
-        Matcher matcher = pattern.matcher(args[index]);
-        if (matcher.find()) {
-            int year = Integer.parseInt(matcher.group(1));
-            int month = Integer.parseInt(matcher.group(2));
-            int day = Integer.parseInt(matcher.group(3));
-            int hour = 0;
-            int minute = 0;
-            double second = 0.0;
-            if (!matcher.group(4).isEmpty()) {
-                hour = Integer.parseInt(matcher.group(5));
-                minute = Integer.parseInt(matcher.group(6));
-                if (!matcher.group(7).isEmpty()) {
-                    second = Double.parseDouble((matcher.group(8)));
-                }
-            }
-            return new Time(year, month, day, hour, minute, second);
+        try {
+            Instant instant = Instant.parse(args[index]);
```

Comment:
This is so much nicer! Obviously, I don't know Java very well! LOL.

Reply:
> I don't know Java very well!

Don't worry about that, you did and are doing just a fantastic contribution to its community already :) And it is a Java 8 thing which should be used in Android with considerations which fortunately isn't a concern here!